libs/gl-client/src/signer/mod.rs

```diff
@@ -761,16 +766,145 @@ impl Signer {
         };
 
         info!("Exiting the signer loop");
         Ok(())
     }
 
+    async fn run_forever_scheduler(
+        &self,
+        mut scheduler: SchedulerClient<tonic::transport::Channel>,
+    ) -> Result<(), anyhow::Error> {
+        let (sender, rx) = mpsc::channel(1);
+        let outbound = ReceiverStream::new(rx);
+        // let inbound_future = scheduler.signer_requests_stream(outbound);
+
+        let mut stream = scheduler
+            .signer_requests_stream(outbound)
+            .await?
+            .into_inner();
+
+        trace!("Starting to stream signer requests from scheduler");
+
+        loop {
+            match stream.message().await {
+                Ok(Some(msg)) => {
+                    let req_id = msg.request_id;
+                    trace!("Processing scheduler request {}", req_id);
+                    match msg.request {
+                        Some(signer_request::Request::ApprovePairing(req)) => {
+                            self.process_pairing_approval(req_id, req, sender.clone())
+                                .await
+                        }
+                        None => {
+                            debug!("Received an empty signing request");
+                        }
+                    };
+                }
+                Ok(None) => {
+                    debug!("End of stream, this should not happen by the server");
+                    return Err(anyhow!("Scheduler closed the stream"));
+                }
+                Err(e) => {
+                    debug!("Got an error from the scheduler {}", e);
+                    return Err(anyhow!("Scheduler stream error {e:?}"));
+                }
+            };
+        }
+    }
+
+    async fn process_pairing_approval(
+        &self,
+        req_id: u32,
+        req: ApprovePairingRequest,
+        stream: mpsc::Sender<SignerResponse>,
+    ) -> () {
+        let mut data = vec![];
+        data.put(req.session_id.as_bytes());
+        data.put_u64(req.timestamp);
+        data.put(&req.node_id[..]);
+        data.put(req.device_name.as_bytes());
+        data.put(req.restrs.as_bytes());
+
+        // Check that the signature matches
+        let pk = UnparsedPublicKey::new(&ECDSA_P256_SHA256_FIXED, req.pubkey.clone());
+        if pk.verify(&data, &req.sig).is_err() {
+            debug!("Got an invalid signature processing pairing approval");
+            return;
+        }
+
+        // Decode rune as we expect it to be bytes in the pending request.
+        let rune = match general_purpose::URL_SAFE.decode(req.rune.clone()) {
+            Ok(r) => r,
+            Err(e) => {
+                debug!("Could not decode rune processing pairing approval {}", e);
+                return;
+            }
+        };
+
+        // Check that the rune matches
+        match self.verify_rune(PendingRequest {
+            request: vec![],
+            uri: "/cln.Node/ApprovePairing".to_string(),
+            signature: req.sig,
```

Comment:
Does the Context need to be updated to handle the signature?
https://github.com/Blockstream/greenlight/blob/e38a37613da7558c853f24be700c193f194a6bc9/libs/gl-client/src/runes.rs#L159